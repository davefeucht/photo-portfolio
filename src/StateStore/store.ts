import { configure, makeAutoObservable } from 'mobx';
import { ImageData, Image, Store, Category, Post, Page } from '../utils/types';

configure({ enforceActions: 'observed' });

class stateStore implements Store {
    menuState = 'closed'
    applicationRoot: HTMLElement = null;
    screenInfo: {
        width: number,
        height: number
    }
    siteInfo: {
        siteName: string,
        siteUrl: string
    }
    visiblePost: {
        postId: number,
        postTitle: string,
        tags: number[],
        tagNames: string[],
        fullImageUrl: string,
        width: number,
        height: number
    };
    categoryList: Category[] = [];
    currentCategoryPosts: Post[] = [];
    currentCategoryData: Category = {
        id: 0,
        count: 0,
        description: '',
        link: '',
        name: '',
        slug: '',
        taxonomy: '',
        parent: 0
    };
    pages: Page[] = [];
    currentPageData: Page = {
        id: 0,
        link: '',
        modified: '',
        slug: '',
        type: '',
        title: {
            rendered: ''
        },
        content: {
            rendered: '',
            protected: false
        },
        author: 0,
        featured_media: 0
    };
    maxItemsPerPage = 10;

    constructor() {
        makeAutoObservable(this);
    }

    setMenuState = (state: string) => {
        this.menuState = state;
    }

    toggleMenuState = () => {
        this.menuState = this.menuState === 'closed' ? 'open' : 'closed';
    }

    setApplicationRoot = (element: HTMLElement) => {
        this.applicationRoot = element;
    }

    setSiteName = (name: string) => {
        this.siteInfo.siteName = name;
    }

    setVisiblePost = (postId: number, postTitle: string) => {
        this.visiblePost.postId = postId;
        this.visiblePost.postTitle = postTitle;
    }

    setVisiblePostImage = (fullImageUrl: string) => {
        this.visiblePost.fullImageUrl = fullImageUrl;
    }

    setVisiblePostTags = (tagNames: string[]) => {
        tagNames.forEach(tagName => {
            this.visiblePost.tagNames.push(tagName);
        });
    }

    clearVisiblePostTagNames = () => {
        this.visiblePost.tagNames.splice(0, this.visiblePost.tagNames.length);
    }

    setCategoryList = (categories: Category[]) => {
        this.categoryList.length = 0;
        categories.forEach((category, index) => {
            this.categoryList[index] = category;
        })
    }

    setCategoryPosts = (posts: Post[]) => {
        this.currentCategoryPosts.length = 0;
        posts.forEach((post, index) => {
            this.currentCategoryPosts[index] = post;
        })
    }

    setCategoryData = (categoryData: Category) => {
        Object.keys(categoryData).forEach((key: keyof Category) => {
            this.currentCategoryData[key] = categoryData[key];
        });
    }

    setThumbnailImageUrl = (imageData: ImageData) => {
        this.currentCategoryPosts[imageData.post_index].thumbnail_image = imageData.image_url;
    }

    setCurrentPost = (postData: Post) => {
        Object.keys(postData).forEach((key: keyof Image) => {
            this.visiblePost[key] = postData[key];
        })
    }

    setPages = (pages: Page[]) => {
        pages.forEach(page => {
            this.pages.push(page);
        });
    }

    setPageData = (pageData: Page) => {
        Object.assign(this.currentPageData, pageData);
    }
}

export default stateStore;
