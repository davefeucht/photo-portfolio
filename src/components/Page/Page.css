.page {
  min-height: 100%;
  display: grid;
  grid-template-columns: 100%;
  grid-template-areas: "page-header" "page-content";
  grid-template-rows: .25fr 4fr;
}