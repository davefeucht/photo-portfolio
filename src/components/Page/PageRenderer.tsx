import PageContent from 'components/PageContent/PageContent';
import SectionHeader from 'components/SectionHeader/SectionHeader';
import * as React from 'react';

interface Props {
    title: string;
    content: string;
}

const PageRenderer: React.FC<Props> = ({
    title,
    content
}) => {
    return (
        <div className="page">
            <SectionHeader title={title} />
            <PageContent content={content} />
        </div>
    );
};

export default PageRenderer;
