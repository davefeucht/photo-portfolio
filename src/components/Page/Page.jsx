/** ***************
* Page component implements displaying a page in the application
**************** */

import './Page.css';

import { observer } from 'mobx-react';
import PropTypes from 'prop-types';
import React, { useEffect } from 'react';
import { useParams } from 'react-router-dom';

import { getPage } from '../../utils/Api.js';
import PageContent from '../PageContent/PageContent.jsx';
import SectionHeader from '../SectionHeader/SectionHeader.jsx';

function Page({ stateStore }) {
    const { pageId } = useParams();
    const { currentPageData } = stateStore;

    useEffect(() => {
        getPage(pageId, stateStore.siteInfo.siteUrl)
            .then(page => {
                stateStore.setPageData(page);
            });
    }, [pageId]);

    return (
        <div className="page">
            <SectionHeader title={currentPageData.title?.rendered} />
            <PageContent content={currentPageData.content?.rendered} />
        </div>
    );
}

Page.displayName = 'Page';

Page.propTypes = {
    stateStore: PropTypes.object.isRequired
};

export default observer(Page);
