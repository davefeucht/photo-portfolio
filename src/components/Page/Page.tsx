/** ***************
* Page component implements displaying a page in the application
**************** */

import './Page.css';

import { observer } from 'mobx-react';
import * as React from 'react';
import { useEffect } from 'react';
import { useParams } from 'react-router-dom';

import { getPage } from '../../utils/Api';
import {
    Page as PageState,
    SiteInfo
} from '../../utils/types';
import PageContent from '../PageContent/PageContent';
import SectionHeader from '../SectionHeader/SectionHeader';

interface PageProps {
    currentPageData: PageState,
    siteInfo: SiteInfo,
    setPageData: (a: PageState) => void
}

const Page: React.FC<PageProps> = ({ currentPageData, siteInfo, setPageData }) => {
    const { pageId } = useParams();

    useEffect(() => {
        if (pageId) {
            getPage(parseInt(pageId), siteInfo.siteUrl)
                .then(page => {
                    setPageData(page);
                });
        }
    }, [pageId]);

    return (
        <div className="page">
            <SectionHeader title={currentPageData.title?.rendered} />
            <PageContent content={currentPageData.content?.rendered} />
        </div>
    );
};

Page.displayName = 'Page';

export default observer(Page);
