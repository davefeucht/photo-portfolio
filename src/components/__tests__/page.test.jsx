import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import StateStore from '../../StateStore/store';
import Page from '../Page/Page';

const page = {
    id: 150,
    content: {
        rendered: 'Contact me!'
    },
    title: {
        rendered: 'Contact'
    }
};

jest.mock('../../utils/Api', () => ({
    getPage: () => Promise.resolve(page)
}));

test('Page displays', () => {
    const store = new StateStore();
    const component = renderer.create(
        <MemoryRouter initialEntries={['/page/150']}>
            <Page stateStore={store} />
        </MemoryRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
