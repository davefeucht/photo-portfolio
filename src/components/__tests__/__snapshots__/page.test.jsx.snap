// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page displays 1`] = `
<div
  className="page"
>
  <div
    className="section-header"
  >
    <div
      className="title"
    />
  </div>
  <div
    className="page__content"
    dangerouslySetInnerHTML={
      {
        "__html": "",
      }
    }
  />
</div>
`;
